src/assessments/seo/keyphraseLengthAssessment.js

```diff
@@ -1,31 +1,99 @@
-var AssessmentResult = require( "../../values/AssessmentResult.js" );
-
+let Assessment = require( "../../assessment.js" );
+let AssessmentResult = require( "../../values/AssessmentResult.js" );
+const inRange = require( "lodash/inRange" );
+const merge = require( "lodash/merge" );
 /**
- * Assesses the keyphrase presence and length
- *
- * @param {Paper} paper The paper to use for the assessment.
- * @param {Researcher} researcher The researcher used for calling research.
- * @param {Jed} i18n The object used for translations
- * @returns {AssessmentResult} The result of this assessment
-*/
-function keyphraseAssessment( paper, researcher, i18n ) {
-	var keyphraseLength = researcher.getResearch( "keyphraseLength" );
-
-	var assessmentResult = new AssessmentResult();
-
-	if ( ! paper.hasKeyword() ) {
-		assessmentResult.setScore( -999 );
-		assessmentResult.setText( i18n.dgettext( "js-text-analysis", "No focus keyword was set for this page. " +
-			"If you do not set a focus keyword, no score can be calculated." ) );
-	} else if ( keyphraseLength > 10 ) {
-		assessmentResult.setScore( 0 );
-		assessmentResult.setText( i18n.dgettext( "js-text-analysis", "The keyphrase is over 10 words, a keyphrase should be shorter." ) );
+ * Assessment to check whether the keyphrase has a good length.
+ */
+class KeyphraseLengthAssessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		let defaultConfig = {
+			scores: {
+				keyphraseLength: 0,
+			},
+			assessment: {
+				recommendedMaximum: 5,
```

Comment:
I think that these values are strictly speaking not correct; the recommended maximum is actually 4, while 5 is already over the recommended maximum. The same holds true for the acceptable maximum. This also leads to the strange situation that you actually have to deduct 1 to provide the recommended values in the feedback string.

I'd recommend to change these values to 4 and 8, respectively, and to change the if conditions accordingly. (Have a look at `src/helpers/inRange.js`, there might be some suitable functions for this.)